applications/plugins/Geomagic/src/GeomagicDriver.cpp

```diff
@@ -254,10 +260,22 @@ void GeomagicDriver::initDevice()
     m_hHD = hdInitDevice(d_deviceName.getValue().c_str());
 
     if (HD_DEVICE_ERROR(error = hdGetError()))
-    {
-        msg_error() << "Failed to initialize the device called " << d_deviceName.getValue().c_str();
-        d_omniVisu.setValue(false);
+    {        
         m_errorDevice = error.errorCode;
+        if (m_errorDevice == 769) // double initialisation, will try to close and reinit device
+        {
+            msg_warning() << "Device has already been initialized. Will clear driver and reinit the device properly.";
+            m_hHD = hdGetCurrentDevice();
+            if (m_hHD != UINT_MAX)
+            {
+                clearDevice();
+                return initDevice();
```

Comment:
Risks of infinite recursion?

Reply:
normally no because we clear the device so it should not enter anymore in the "already init" block.
But I can add a security check.

Reply:
done, @guparan could you review my last commit and set to Ready if ok. Use 10 as magic number.